src/SignalR/clients/ts/signalr/src/HubConnection.ts

```diff
@@ -294,8 +294,17 @@ export class HubConnection {
             return this._stopPromise!;
         }
 
+        const state = this._connectionState;
         this._connectionState = HubConnectionState.Disconnecting;
 
+        if (state === HubConnectionState.Connected) {
+            await this._sendWithProtocol(this._createCloseMessage());
+            // Transport can close during the above 'await', we should check if that occurred and noop.
+            if (this._connectionState !== HubConnectionState.Disconnecting) {
+                return this._stopPromise!;
+            }
+        }
+
         this._logger.log(LogLevel.Debug, "Stopping HubConnection.");
 
         if (this._reconnectDelayHandle) {
```

Comment:
We should only try sending a close message after we verify we're not in a reconnect delay which would mean there's no active transport.